test/creditguild-test.js

```diff
@@ -0,0 +1,206 @@
+const { ethers } = require("hardhat");
+const { expect } = require("chai");
+require("chai").should();
+
+describe("CreditGuild", function () {
+  let MEMBERSHIPFEE = ethers.utils.parseEther("1");
+  let VOUCHAMOUNT = ethers.utils.parseEther("1");
+
+  before(async function () {
+    [ADMIN, ALICE, BOB, TOM, JACK, MEMBER1, MEMBER2, MEMBER3] =
+      await ethers.getSigners();
+
+    const MarketRegistry = await ethers.getContractFactory(
+      "MarketRegistryMock"
+    );
+    const marketRegistry = await MarketRegistry.deploy();
+    const MockERC20 = await ethers.getContractFactory("FaucetERC20");
+
+    // deploy underlyingToken
+    const underlyingToken = await MockERC20.deploy();
+    await underlyingToken.mint(ADMIN.address, ethers.utils.parseEther("100")); // min 100 DAI (underlying)
+    const underlyingAddress = underlyingToken.address;
+
+    // deploy uToken
+    const uToken = await MockERC20.deploy();
+    await uToken.mint(ADMIN.address, ethers.utils.parseEther("100"));
+    const uTokenAddress = uToken.address;
+    
+    // fund members' accounts
+    await underlyingToken.mint(MEMBER1.address, ethers.utils.parseEther("100")); // min 100 DAI (underlying)
+    await underlyingToken.mint(MEMBER2.address, ethers.utils.parseEther("100")); // min 100 DAI (underlying)
+    await underlyingToken.mint(MEMBER3.address, ethers.utils.parseEther("100")); // min 100 DAI (underlying)
+
+    // deploy userManager
+    const UserManager = await ethers.getContractFactory("UserManagerMock");
+    const userManager = await UserManager.deploy();
+    await userManager.setIsMember(true);
+    // await userManager.getVouchingAmount();
+
+    // add mockUtoken contract for the mock underlying token (DAI)
+    await marketRegistry.addUToken(underlyingAddress, uTokenAddress);
+    await marketRegistry.addUserManager(underlyingAddress, userManager.address);
+
+    // deploy creditGuild
+    const CreditGuild = await ethers.getContractFactory("CreditGuild");
+    creditGuild = await CreditGuild.deploy(
+      MEMBERSHIPFEE,
+      VOUCHAMOUNT,
+      marketRegistry.address,
+      uTokenAddress,
+      underlyingAddress
+    );
+
+  });
+
+
+  describe("Initialize", function () {
+    it("Should initialize first 3 members", async function () {
+
+      const tx = await creditGuild.initialize(
+        MEMBER1.address,
+        MEMBER2.address,
+        MEMBER3.address
+      );
+
+      const resp = await tx.wait();
+      result = resp.events[resp.events.length - 1].args.initialMembers;
+
+      array2 = [MEMBER1.address, MEMBER2.address, MEMBER3.address]
+
+      expect(result[0]).to.equal(array2[0]);
+      expect(result[1]).to.equal(array2[1]);
+      expect(result[2]).to.equal(array2[2]);
+
```

Comment:
You should also check these address are now members. You could write this test like this.

```js
const memberAddresses = [MEMBER1.address, MEMBER2.address, MEMBER3.address];
const tx = await creditGuild.initialize(...memberAddresses);
const resp = await tx.wait();
const initialMembers = resp.events[resp.events.length - 1].args.initialMembers;
for (let i = 0; i < memberAddresses.length; i++) {
      expect(initialMembers[i]).eq(memberAddresses[i]);
      expect(await creditGuild.checkIsMember(memberAddresses[i])).eq(true);
}
```